### ISSUE TITLE
Edit Marker to enable moving a marker location

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/land-explorer-front-end#317]
**Describe the bug**
In the Drawing Tools menu of LX the Edit option allows the user to edit the points in a line or a poly. However when you click an existing marker you can't do anything. It would make sense to also be able to edit the location of a marker.

**To Reproduce**
1. Ensure there is an existing drawn marker on your map.
2. Go to Drawing Tools
3. Click on Edit
4. Go to an existing marker and click on it
5. You can't change the location.
Note - this behaviour is different to Lines or Polys so feels like a bug.

**Expected behaviour**
When you click on a marker after clicking edit you should be able to drag the marker to a new location

**Comment by lin-d-hop** [comment]
After some preliminary investigations...
Mapbox gl doesn't allow markers to be moved.
There is a chance we could implement markers via the mapbox gl 'feature' functionality, however this might break our clustering implementation

Quick spike to see what happens.
If anything breaks, let's update the issue here and close it.

**Comment by ms0ur1s** [comment]
Getting this at the moment:

![Image](https://github.com/DigitalCommons/land-explorer-front-end/assets/22494228/ffabe60b-7a6a-4558-b690-6832438f9bf6)

@rogup is the below code sufficient for a quick test?
The correct way, I guess, would be to create a bespoke feature from scratch more akin to our polygon and lines.

![Image](https://github.com/DigitalCommons/land-explorer-front-end/assets/22494228/210cd426-6309-404d-a64c-22d6c6c33578)

**Comment by ms0ur1s** [comment]
@lin-d-hop, I'll wait to see what @rogup says about the above before closing.

**Comment by rogup** [comment]
@ms0ur1s I think that error is because an expected prop hasn't been given, so I don't know if it's using Features that is necessarily the problem.

I had another quick look at how polygons and lines are editable and can see that we use the `react-mapbox-gl-draw` drawControl component. So I think we don't actually need to implement our own onDragStart/End methods, and could do the same for markers.

I think it does seem doable to display markers in this way, as a GeoJSON layer of Point features, and then feed in the features to drawControl. This would make how we display markers, polys and lines more consistent. I found this library too, which I did a really quick test with, and it looks like clustering will still work https://www.npmjs.com/package/react-mapbox-gl-cluster

The branch I played with: https://github.com/DigitalCommons/land-explorer-front-end/commit/09d082d88054d2e2e492c3ad533e0ae1d1285d67

So I think making markers moveable is possible, but wouldn't be a really quick fix (it would probably take 1-2 days)

**Comment by ms0ur1s** [comment]
@rogup, thanks man. Funnily enough, when I was doing a little digging in the react-mapbox-gl documentation I had a similar though about the GeoJSON layer of Point features. The cluster library is a great find.

If @lin-d-hop is happy to go ahead, do you think these changes are within my capabilities, if I use your initial work as a starting point and make sure give you a shout when I encounter any difficulties?

**Comment by rogup** [comment]
Yeah I think it's something that you can do :) And learning about how stuff is rendered in Mapbox is probably a useful thing, since it relates to your frontend UI expertise and might be related to work we do soon on Mykomap too.

Happy to answer any questions or jump on a call to give any guidance. The structure of data in GeoJSON and the way that feature IDs work is something that I have been working on a lot recently (with property boundaries)

**Comment by ms0ur1s** [comment]
Thanks @rogup for all your help and patience, as usual. 😀

@lin-d-hop, in light of Rohit's comments are you happy for me to carry on with this as a dev ticket?

**Comment by ms0ur1s** [comment]
Thanks @lin-d-hop, understood. Also, if I get any sense that I'm hitting a brick wall I reach out to @rogup.

**Comment by lin-d-hop** [comment]
Cool, yes, rule of thumb - don't be stuck for an hour. Reach out!

**Comment by lin-d-hop** [comment]
Closing for now as the work on this will change with #327 and this ticket is referenced there
